packages/status-core/src/utils/public-key-to-color-hash.ts

```diff
@@ -0,0 +1,91 @@
+// TODO?: import only used; or just '@noble/secp256k1' instead
+import * as secp256k1 from 'ethereum-cryptography/secp256k1'
+
+type ColorHash = number[][]
+
+const COLOR_HASH_COLORS_COUNT = 32
+const COLOR_HASH_SEGMENT_MAX_LENGTH = 5
+
+export function publicKeyToColorHash(publicKey: string): ColorHash | undefined {
+  const publicKeyHex = publicKey.replace(/^0[xX]/, '') // ensures hexadecimal digits without "base prefix"
+
+  let compressedPublicKeyDigits: string
+  try {
+    // TODO?: consider obtaining without a crypto lib
+    compressedPublicKeyDigits =
+      secp256k1.Point.fromHex(publicKeyHex).toHex(true) // validates and adds "sign prefix" too
```

Comment:
Non-blocking: If size of this would become an issue, we could just rewrite/copy-paste relevant parts (i.e. validation, splitting and prefixing).